nototools/android_patches.py

```diff
@@ -0,0 +1,431 @@
+
+  Make 12 characters default to color emoji style.
+
+  The Noto Sans Symbols font and the script that subsets it have been
+  updated to excluce the 12 characters we decided to make default to
+  color in Android.
+
+  The Noto Sans Simplified Chinese font was also subsetted.
+
+  (5855164ed87)"""
+
+  def _remove_from_cmap(infile, outfile, exclude=[]):
+    font = ttLib.TTFont(infile)
+    font_data.delete_from_cmap(font, exclude)
+    font.save(outfile)
+
+  EMOJI = (
+      [0x26BD, 0x26BE, 0x1F18E]
+      + range(0x1F191, 0x1F19A+1)
+      + [0x1F201, 0x1F21A, 0x1F22F]
+      + range(0x1F232, 0x1F236+1)
+      + [0x1F238, 0x1F239, 0x1F23A, 0x1F250, 0x1F251]
+  )
+
+  names = ['cjk/NotoSans%sCJK%s-Regular.otf' % (m, v)
+           for m in ['', 'Mono']
+           for v in ['jp', 'kr', 'sc', 'tc']]
+  for font_name in names:
+    print 'remove cjk emoji', font_name
+    _remove_from_cmap(
+        font_name,
+        path.join(outdir, font_name),
+        exclude=EMOJI)
+
+
+def _fix_arabic_ligatures():
+  """Fix U+FDF2 and Allah-related ligatures in Naskh fonts.
+
+  U+FDF2 ARABIC LIGATURE ALLAH ISOLATED FORM missed the initial alef in
+  its glyph, while the Allah ligatures were incorrectly formed.
+
+  (8ae3a28c050ef9)"""
+
+  # noto-fonts/issues/384 was fixed, so this patch is not needed.
+  pass
+
+
+# below are used by _subset_symbols
+
+# Unicode blocks that we want to include in the font
+BLOCKS_TO_INCLUDE = """
+20D0..20FF; Combining Diacritical Marks for Symbols
+2100..214F; Letterlike Symbols
+2190..21FF; Arrows
+2200..22FF; Mathematical Operators
+2300..23FF; Miscellaneous Technical
+2400..243F; Control Pictures
+2440..245F; Optical Character Recognition
+2460..24FF; Enclosed Alphanumerics
+2500..257F; Box Drawing
+2580..259F; Block Elements
+25A0..25FF; Geometric Shapes
+2600..26FF; Miscellaneous Symbols
+2700..27BF; Dingbats
+27C0..27EF; Miscellaneous Mathematical Symbols-A
+27F0..27FF; Supplemental Arrows-A
+2800..28FF; Braille Patterns
+2A00..2AFF; Supplemental Mathematical Operators
+"""
+
+# One-off characters to be included, needed for backward compatibility and
+# supporting various character sets, including ARIB sets and black and white
+# emoji
+ONE_OFF_ADDITIONS = {
+    0x27D0, # ⟐ WHITE DIAMOND WITH CENTRED DOT
+    0x2934, # ⤴ ARROW POINTING RIGHTWARDS THEN CURVING UPWARDS
+    0x2935, # ⤵ ARROW POINTING RIGHTWARDS THEN CURVING DOWNWARDS
+    0x2985, # ⦅ LEFT WHITE PARENTHESIS
+    0x2986, # ⦆ RIGHT WHITE PARENTHESIS
+    0x2B05, # ⬅ LEFTWARDS BLACK ARROW
+    0x2B06, # ⬆ UPWARDS BLACK ARROW
+    0x2B07, # ⬇ DOWNWARDS BLACK ARROW
+    0x2B24, # ⬤ BLACK LARGE CIRCLE
+    0x2B2E, # ⬮ BLACK VERTICAL ELLIPSE
+    0x2B2F, # ⬯ WHITE VERTICAL ELLIPSE
+    0x2B56, # ⭖ HEAVY OVAL WITH OVAL INSIDE
+    0x2B57, # ⭗ HEAVY CIRCLE WITH CIRCLE INSIDE
+    0x2B58, # ⭘ HEAVY CIRCLE
+    0x2B59, # ⭙ HEAVY CIRCLED SALTIRE
+}
+
+# letter-based characters, provided by Roboto
```

Comment:
Add a TODO here to see if we need to change this subset based on Noto Serif coverage (so the serif fallback chain would support them).